ledger/ledger-utxo/src/main/java/net/corda/v5/ledger/utxo/StateRef.java

```diff
@@ -74,21 +70,17 @@ public int getIndex() {
      */
     public static StateRef parse(@NotNull final String value, DigestService digestService) {
         try {
-            if (!value.contains(DELIMITER)) {
-                throw new IllegalArgumentException(
-                        MessageFormat.format("Failed to parse a StateRef from the specified value. The value: {0} doesn't contain expected delimiter (:)", value)
-                        );
-            }
-            if (value.length() < MINIMUM_HASH_LENGTH) {
+            final int lastIndexOfDelimiter = value.lastIndexOf(DELIMITER);
+            if (lastIndexOfDelimiter == -1) {
```

Comment:
I don't think this does quite the right thing - a `SecureHash` contains a delimiter, but would not be a valid `StateRef`.

Technically if we did not have this check, then I think it would fail the index parsing anyway (since it would try parsing a hash, not an int). But if you want a slightly better error message, then we should probably check for exactly 2 instances of the delimiter - the first would be part of the `SecureHash`, the second would separate this from the index.

Reply:
Checking the `SecureHash` delimiter in this class mixes the two classes slightly. StateRefs should not need to know the internal structure of `SecureHash`es.
Theoretically, we should only check that we have a `:` and a number after it.
The `SecureHash` class should check whether the first part is a proper SecureHash or not.
But the theory part would be only deal-breaker if we had any chances to have SecureHashes with more than one or zero `:`s.
On the other hand, I think the `lastIndexOf()` solution is simpler and cleaner slightly than checking two delimiters.